strongbox-web-core/src/test/java/org/carlspring/strongbox/rest/ArtifactControllerTest.java

```diff
@@ -39,66 +36,49 @@
-import org.springframework.context.annotation.Import;
-import org.springframework.http.MediaType;
-import org.springframework.security.authentication.AuthenticationServiceException;
-import org.springframework.security.test.context.support.WithUserDetails;
-import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-import org.springframework.test.context.web.WebAppConfiguration;
 import static com.jayway.restassured.module.mockmvc.RestAssuredMockMvc.given;
-import static org.apache.http.HttpStatus.SC_FORBIDDEN;
-import static org.apache.http.HttpStatus.SC_UNAUTHORIZED;
 import static org.carlspring.maven.commons.util.ArtifactUtils.getArtifactFileName;
 import static org.carlspring.strongbox.testing.TestCaseWithArtifactGeneration.generateArtifact;
-import static org.hamcrest.Matchers.containsString;
 import static org.junit.Assert.*;
 
 /**
- * Created by yury on 8/3/16.
+ * Test cases for {@link ArtifactController}.
+ *
+ * @author Alex Oreshkevich, Martin Todorov
  */
-
-// testMetadataAtVersionLevel doesn't work
-// Artifact org/carlspring/strongbox/metadata/metadata-foo/3.1-SNAPSHOT/maven-metadata.xml not found
-
-// maven-metadata.xml was not generated properly
 @RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration(classes = WebConfig.class)
-@WebAppConfiguration
+@RestAssuredTest
 public class ArtifactControllerTest
-        extends BackendBaseTest
+        extends RestAssuredBaseTest
 {
 
     public static final String PACKAGING_JAR = "jar";
 
     private static final String TEST_RESOURCES = "target/test-resources";
 
-    private static final File GENERATOR_BASEDIR = new File(ConfigurationResourceResolver.getVaultDirectory() + "/local");
+    private static final File GENERATOR_BASEDIR = new File(ConfigurationResourceResolver.getVaultDirectory() +
+                                                           "/local");
 
     private static final File REPOSITORY_BASEDIR_RELEASES = new File(ConfigurationResourceResolver.getVaultDirectory() +
                                                                      "/storages/storage0/releases");
 
-    @Configuration
-    @Import({ StorageIndexingConfig.class,
-              StorageApiConfig.class,
-              CommonConfig.class,
-              ClientConfig.class,
-              DataServiceConfig.class
-            })
-    public static class SpringConfig { }
-
     private MetadataMerger metadataMerger;
 
 
-    @Before
-    public void setUpClass()
+    @BeforeClass
+    public static void setUpClass()
             throws Exception
     {
+        // to make this test idempotent (in general) we will check and remove previous generated artifacts if they are present
+        // notice that we can't delete the whole repository basedir because it contains also .index, .trash etc.
+        removeDir(REPOSITORY_BASEDIR_RELEASES.getAbsolutePath() + "/org");
```

Comment:
@AlexOreshkevich ,

I understand what you mean, but this is not a good idea. My reasoning is this: at some point in the very near future, I would like to be able to run the tests in parallel (see [SB-320](https://dev.carlspring.org/youtrack/issue/SB-320). If you delete these directories each time the `setUpClass` is invoked before each test, then some of the running test methods might fail, if they still need the resources.

I've been very particular and put specific comments next to each `generateArtifact` statement so that it is absolutely clear which test method is using the respective generated artifact. None of the test's methods are using each other's resources in any way.